src/styles/main.scss

```diff
@@ -1,3 +1,70 @@
-body {
-  margin: 0;
+$cell-size: 100px;
+$border-color: black;
+$border-width: 1px;
+$number-size: 30px;
+$gap: 1px;
+$padding: 10px;
+$days-in-week: 7;
+$animation-duration: 0.5s;
+$start-days: mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6;
+
+.main-container {
+  display: flex;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($cell-size + $border-width + $gap) * $days-in-week + 2 * $padding;
+  padding: $padding;
+  gap: $gap;
+  background-color: white;
+  cursor: default;
+
+  @each $day, $order in $start-days {
+    &--start-#{$day} > :first-child {
+      margin-left: $order * ($cell-size + $border-width * 2 + $gap);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} {
+      .calendar__cell:nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__cell {
+    width: $cell-size;
+    height: $cell-size;
+    border: $border-width solid $border-color;
+    background-color: #eee;
```

Comment:
It is a good practice to make all colors [variable](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)